spec: Propses specification for NFC topic /validators/nfc

mikaelbr opened the pull request:
# Proposal: NFC

Adds initial specification for topic sending travelcard and optional token from reading NFC card.

## Context and known limitations

This will in cases where read access key is not distributed to devices, not contain tokens.
In that case we should use APDU flow. See separate spec proposal

---

## Checklist

- [x] Includes documentation file with description, topics and examples.
- [x] Includes JSON Schema
- [x] Includes passing and failing tests.

## Confirmed and verified by parties

- [ ] MT Buss
- [ ] Consat
- [ ] Fara

mikaelbr committed:
spec: propose specification /validators/nfc

mikaelbr committed:
fix: typo NDS -> NSD

mikaelbr committed:
fix: issue with required eventTimestamp for nfc

mikaelbr committed:
fix: description typos

mikaelbr commented:
Det vil være tilfeller hvor det er presentert ugyldig/usupportert kort. flytdiagram må vise direkte avvisning

Topguy commented on specifications/validators/nfc/nfc.schema.json:
```diff
@@ -0,0 +1,38 @@
+      "$id": "#/properties/travelCardNumber",
+      "type": "string",
+      "pattern": "^\\d{9}$",
+      "description": "Serial number for travel card, 9 digits as string."
```
Dette er i praksis en 32bit integer så alle lengder ifra 1-10 er strengt tatt gyldige.

mikaelbr replied:
Er det representert annerledes på kort enn i systemene ellers? Jeg mener det er enkelte som har 0-padded tall? Det var i alle fall noen som sendte inn det i et system i alle fall, men mulig det var feil innsendt serienummer

mikaelbr commented:
Endre slik at det gjøres en komplett dump av hele kortet som sendes videre.

mikaelbr committed:
fix: laxes travel card validation rule

mikaelbr committed:
fix: adds supported card to flow diagram

mikaelbr committed:
fix: adds supported card to flow diagram

feat: adds smart card content dump to nfc spec

fix: required fields for cardContent

docs: adds docs on card contnet

docs: adds note on several travel rights

fix: adds type desfire

fix: changes from desfire to nsd

sijangurung commented:
👍 MtBus

mikaelbr commented on specifications/validators/nfc/nfc.example.json:
```diff
@@ -0,0 +1,14 @@
+  "cardContent": [
+    {
+      "appId": "<ID>",
+      "type": "desfire",
```
```suggestion
      "type": "nsd",
```

mikaelbr commented:
Add configure/initialize if read token should dump entire card.

extend flow diagram to reflect configure set

mikaelbr committed:
chore: update types

mikaelbr committed:
docs: adds card dump to flow chart
